ansible/action_plugins/agnosticd_odcr.py

```diff
@@ -133,6 +139,72 @@ def check_response(response):
         display.error(pp.pformat(response))
         six.raise_from(AnsibleError("Error requesting EC2 offerings."), err)
 
+def reservation_match(r1, r2):
+    """Return true if 2 reservations match.
+
+    Reservations match when all the reservations keys except instance_count
+    are identical"""
+
+    # same ref
+    if r1 == r2:
+        return True
+
+    compared = set(r1.keys()).union(set(r2.keys())).difference(set(['instance_count']))
+
+    for c in compared:
+        if c in r1 and c not in r2:
+            return False
+
+        if c not in r1 and c in r2:
+            return False
+
+        if c not in r1 and c not in r2:
+            continue
+
+        if c in r1 and c in r2:
+            if r1[c] == r2[c]:
+                continue
+
+            return False
+
+    return True
+
+
+def inject_reservation(l, *reservations):
+    """Inject reservations by incrementing the count if they match
+
+    Return the list with the reservation injected, leave the original list
+    passed as argument untouched."""
+
+    l = deepcopy(l)
+
+    for reservation in deepcopy(reservations):
+        for r in l:
+            if reservation_match(r, reservation):
+                r['instance_count'] = int(r['instance_count']) + int(reservation['instance_count'])
+                break
+            else:
+                continue
+            break
```

Comment:
This break statement will never be hit I think? Can this logic be simplified to:
```
        for r in l:
            if reservation_match(r, reservation):
                r['instance_count'] = int(r['instance_count']) + int(reservation['instance_count'])
                break
```